content/400-reference/200-api-reference/100-prisma-schema-reference.mdx

```diff
@@ -19,6 +19,7 @@ A `datasource` block accepts the following fields:
 | `provider`          | **Yes**  | String (`postgresql`, `mysql`, `sqlite`, `sqlserver`, `mongodb`, `cockroachdb`) | Describes which data source connectors to use.                                                                                                                                                                                                                                                                                                    |
 | `url`               | **Yes**  | String (URL)                                                                    | Connection URL including authentication info. Most connectors use [the syntax provided by the database](/reference/database-reference/connection-urls#format).                                                                                                                                                                                    |
 | `shadowDatabaseUrl` | No       | String (URL)                                                                    | Connection URL to the shadow database used by Prisma Migrate. Allows you to use a cloud-hosted database as the shadow database.                                                                                                                                                                                                                   |
+| `directUrl`         | No       | String (URL)                                                                    | Connection URL for direct connection to the database.<br /><br />If you use the Data Proxy, the generated Prisma Client uses the URL defined by the `url` argument. All other connections use the URL defined by the `directUrl` argument.                                                                                                        |
```

Comment:
Not necessarily Data Proxy specific, also applies to other configurations like PgBouncer (Where you can not run Introspection or Migrations).

Reply:
What's a good term for these other configurations in general? 'If you use the Data Proxy or...'

Reply:
Maybe:
> Some Prisma commands require a direct connection to the database, so Prisma Data Proxy or PgBouncer can not be used for e.g. Introspection or Migration commands.

Reply:
How about:

If you use a connection pooler URL in the `url` argument (such as Data Proxy, pgBouncer), Prisma CLI commands that require a direct connection to the database (such as `migrate` or `db pull`) use the URL in the `directUrl` argument.

Reply:
Gone for a slight variant:

> If you use a connection pooler URL in the `url` argument (for example, if you use the Data Proxy or pgBouncer), Prisma CLI commands that require a direct connection to the database (such as `migrate` or `db pull`) use the URL in the `directUrl` argument. 

and added a link to the Data Platform docs